### PR TITLE
Generated chain details

### PR DESCRIPTION
**Pull request by dimkouv**
Generates `generated_chains.go` file after running `go generate`.
A github workflow is added that marks the pr 🔴 in-case the generated files are not updated.

The generated files provides the following:
```
// easily get chain information by statically defined chain name
chainselectors.AVALANCHE_MAINNET.EvmChainID

// get all the chains
chainselectors.ALL
```

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -17,4 +17,10 @@ jobs:
       - name: Build
         run: go build -v ./...
       - name: Test
-        run: go test -v ./...
\ No newline at end of file
+        run: go test -v ./...
+      - name: Make sure generated files are updated
```
Comment (mateusz-sekara) [inline .github/workflows/go.yml:21]: Shouldn't that be run before `Test` phase (and maybe `Build`)?

Comment (mateusz-sekara) [reply .github/workflows/go.yml:21]: Also worth adding to readme that `go generate` is required for local development.

File: generated_chains.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	ETHEREUM_TESTNET_SEPOLIA_ARBITRUM_1,
+	ETHEREUM_TESTNET_SEPOLIA_BASE_1,
+	ETHEREUM_TESTNET_SEPOLIA_KROMA_1,
+	ETHEREUM_TESTNET_SEPOLIA_OPTIMISM_1,
+	ETHEREUM_TESTNET_SEPOLIA_SCROLL_1,
+	KAVA_MAINNET,
+	KAVA_TESTNET,
+	POLYGON_MAINNET,
+	POLYGON_TESTNET_MUMBAI,
+	UNNAMED_1000,
```
Comment (mateusz-sekara) [inline generated_chains.go:101]: Hmm, I would say that either `CHAIN_1000` or `TEST_1000` would be better than unnamed. Leaning towards `CHAIN_` prefix

**Review by mateusz-sekara — COMMENTED** [review]
Looks good in general, minor comments

File: generated_chains.go (hunk 2 above)
Comment (makramkd) [reply generated_chains.go:101]: Agree, TEST_ seems appropriate.

**Review by mateusz-sekara — APPROVED** [review]

**Pull request merged**